.travis.yml

```diff
@@ -14,5 +14,28 @@
 
 sudo: required
 services: docker
+
+# Cache intermediate Docker layers. For a description of how this works, see:
+# https://giorgos.sealabs.net/docker-cache-on-travis-and-docker-112.html
+cache:
+  directories:
+    - /home/travis/docker/
+env:
+  global:
+    - DOCKER_CACHE_FILE=/home/travis/docker/cache.tar.gz
+before_install:
```

Comment:
pull out into a bash script.. that is bad enough, no need to frankenstein it with yaml.